### PR TITLE
feat(web): add hash-based tab navigation for back-button and reload support

### PR DESCRIPTION
**Pull request by mennemann**
## Description

Improved UX of AfterTouch by persisting the active tab in the URL hash. This allows for staying on the same tab when reloading and navigating to previous tabs by hitting the back button of the browser.

Previously, reloading returned to the "Overview" tab and the back button always navigated off of AfterTouch. 

https://github.com/user-attachments/assets/241b075c-84a6-4953-aadb-133e146bcd5e

Tested on Chrome and Safari (iOS). Someone should definitely test Firefox before merging.

I am feeling too lazy to fill out the rest right now :)
If you want me to I will fill it out tomorrow though.

## Type of Change

Please check the type of change your PR introduces:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring (no functional changes)
- [ ] Test improvements
- [ ] Build/CI improvements

## Related Issues

- Fixes #[issue number]
- Relates to #[issue number]
- Part of #[issue number]

### API Changes
- [ ] Added new endpoints
- [ ] Modified existing endpoints
- [ ] Added new CLI commands
- [ ] Modified existing CLI commands
- [ ] Added new configuration options

### Implementation Details
- Describe the main changes
- List any new dependencies
- Mention any architectural changes

### Automated Tests
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] All existing tests pass
- [ ] Test coverage maintained or improved

### Manual Testing
- [ ] Tested with real SoundTouch device(s)
- [ ] Tested CLI changes manually
- [ ] Tested in different network environments

**Device(s) tested with:**
- Device model: [e.g. SoundTouch 10]
- Device IP: [e.g. 192.0.2.100]
- Test results: [brief description]

### Test Commands
```bash
# Commands used to test this change
make test
go test ./pkg/client -v -run TestNewFeature
soundtouch-cli --host 192.0.2.100 new-command
```

## Documentation

- [ ] Updated relevant documentation
- [ ] Added code comments for complex logic
- [ ] Updated CLI help text
- [ ] Added usage examples
- [ ] Updated API documentation

**Documentation files updated:**
- [ ] README.md
- [ ] docs/API-Endpoints-Overview.md
- [ ] docs/CLI-REFERENCE.md
- [ ] Code documentation (godoc)

## Backward Compatibility

- [ ] This change is backward compatible
- [ ] This change includes breaking changes (requires major version bump)
- [ ] This change requires configuration migration

**Breaking changes (if any):**
- Describe what breaks
- Provide migration instructions

## Security Considerations

- [ ] No security implications
- [ ] Security review required
- [ ] Added input validation
- [ ] Updated authentication/authorization

## Performance Impact

- [ ] No performance impact
- [ ] Performance improvement
- [ ] Potential performance regression (justify why)

**Performance notes:**
- Measured impact: [benchmarks, timing, memory usage]
- Optimization opportunities: [if any]

## Code Quality

- [ ] Code follows project style guidelines
- [ ] No linting errors
- [ ] No security warnings
- [ ] Memory leaks checked (if applicable)

### Pre-submission Checklist

- [ ] `make check` passes (format, lint, vet)
- [ ] `make test` passes
- [ ] No TODO comments left in production code
- [ ] Error handling is comprehensive
- [ ] Logging is appropriate (not too verbose, not too quiet)

## Deployment Notes

Any special considerations for deployment:
- Configuration changes required
- Database migrations needed
- Service restart required
- Rollback procedures

## Screenshots (if applicable)

If this PR includes UI changes or CLI output changes, include screenshots or terminal output examples.

```bash
# Before
$ soundtouch-cli old-command
Old output...

# After  
$ soundtouch-cli new-command
New improved output...
```

## Additional Notes

Any additional information that reviewers should know:
- Design decisions and trade-offs
- Future work planned
- Alternative approaches considered
- References to external documentation

## Review Requests

**Areas that need special attention:**
- [ ] Error handling logic
- [ ] Performance critical sections
- [ ] Security implications
- [ ] API design choices
- [ ] Documentation clarity

**Specific questions for reviewers:**
1. Question about design choice X?
2. Is error handling sufficient in section Y?
3. Should we consider alternative approach Z?

---

**Reviewer Guidelines:**
- Check that all tests pass
- Verify documentation is updated
- Test manually if device access available  
- Consider backward compatibility
- Evaluate error handling and edge cases

### CODE REVIEW

**Comment by gesellix** [comment]
> I am feeling too lazy to fill out the rest right now :)

That's fine, keep it like that... I have to reduce that... sorry 😬

File: pkg/service/handlers/web/js/script.js [hunk 1]
```diff
@@ -1555,6 +1563,11 @@ document.addEventListener("DOMContentLoaded", () => {
     triggerDiscovery();
     fetchVersion();
 
+    const hash = window.location.hash.slice(1);
+    if (hash && document.getElementById(hash)) {
```
Comment (gesellix) [inline pkg/service/handlers/web/js/script.js]: Do we need some validation that `hash` is a known/existing hash? I'm quite sure that an invalid hash won't break anything, but I'm curious if a check would be best practice.

File: pkg/service/handlers/web/js/script.js [hunk 2]
```diff
@@ -521,6 +522,13 @@ function openTab(evt, tabId) {
     }
 }
 
+window.addEventListener('popstate', function() {
+    const tabId = window.location.hash.slice(1);
+    if (tabId && document.getElementById(tabId)) {
```
Comment (gesellix) [inline pkg/service/handlers/web/js/script.js]: `tabId` or `hash`? ;)
Please use a consistent name, I think `tabId` is fine.

File: pkg/service/handlers/web/js/script.js [hunk 3]
```diff
@@ -509,6 +509,7 @@ function openTab(evt, tabId) {
 
     if (evt) {
         evt.currentTarget.className += " active";
+        history.pushState(null, '', '#' + tabId);
```
Comment (gesellix) [inline pkg/service/handlers/web/js/script.js]: Can we also add the selected speaker for the Migration tab? Yes, I'm a bit greedy... ;)

**Review by gesellix — APPROVED** [review]

File: pkg/service/handlers/web/js/script.js (hunk 1 above)
Comment (mennemann) [reply pkg/service/handlers/web/js/script.js]: Invalid hash just puts you on the "Overview" tab. I think it's fine, although if you want to I can make it such that an invalid hash gets cleared.

File: pkg/service/handlers/web/js/script.js (hunk 2 above)
Comment (mennemann) [reply pkg/service/handlers/web/js/script.js]: hash and tabId are now not exactly the same thing so it should be fine now

File: pkg/service/handlers/web/js/script.js (hunk 3 above)
Comment (mennemann) [reply pkg/service/handlers/web/js/script.js]: Should work now

**Comment by mennemann** [comment]
There is a bug with the Migration tab thing. If you go to Migration and then select a device, you have to press back twice to go to your previous tab, because the history stack looks like this:

```
/tab-migrations?<device_id>
/tab-migrations
/overview
```

I'll fix it tomorrow, so don't merge for now.

Note to self: actively select the placeholder if no device_id is given

File: pkg/service/handlers/web/js/script.js (hunk 3 above)
Comment (mennemann) [reply pkg/service/handlers/web/js/script.js]: We could also do it for data sync, but I think that's unnecessary.

Comment (gesellix) [reply pkg/service/handlers/web/js/script.js]: Yeah, I haven't used data sync for... weeks. :)

File: pkg/service/handlers/web/js/script.js (hunk 1 above)
Comment (gesellix) [reply pkg/service/handlers/web/js/script.js]: Sounds good, overview as fallback is enough. Further navigation resets the wrong tabId.

**Comment by mennemann** [comment]
Should work now

File: pkg/service/handlers/web/js/script.js [hunk 4]
```diff
@@ -1743,7 +1743,10 @@ async function showSummary(deviceId) {
         document.getElementById("migration-summary").style.display = "none";
         return;
     }
-    history.pushState(null, '', '#tab-migration?' + deviceId);
+    const newHash = '#tab-migration?' + deviceId;
+    if (window.location.hash !== newHash) {
```
Comment (gesellix) [inline pkg/service/handlers/web/js/script.js:1747]: This prevents a loop, when switching between different devices.

**Review by gesellix — APPROVED** [review]
LGTM, I only added some minor changes and a non-critical bugfix

**Comment by gesellix** [comment]
Thanks!
